library/kani/src/mem_init.rs

```diff
@@ -331,3 +347,55 @@ fn copy_init_state<const LAYOUT_SIZE: usize, T>(from: *const T, to: *const T, nu
 fn copy_init_state_single<const LAYOUT_SIZE: usize, T>(from: *const T, to: *const T) {
     copy_init_state::<LAYOUT_SIZE, T>(from, to, 1);
 }
+
+#[derive(Clone, Copy)]
+struct ArgumentBuffer {
+    selected_argument: usize,
+    saved_address: *const (),
+    layout_size: usize,
+}
+static mut ARGUMENT_BUFFER: Option<ArgumentBuffer> = None;
+
+#[rustc_diagnostic_item = "KaniResetArgumentBuffer"]
+#[kanitool::disable_checks(pointer)]
+fn reset_argument_buffer() {
+    unsafe { ARGUMENT_BUFFER = None }
+}
+
+#[rustc_diagnostic_item = "KaniStoreArgument"]
+#[kanitool::disable_checks(pointer)]
+fn store_argument<const LAYOUT_SIZE: usize, T>(from: *const T, selected_argument: usize) {
+    if LAYOUT_SIZE == 0 {
```

Comment:
Can we remove this? I'm assuming your transformation should've taken care of this case already, and in the worst case scenario, this shouldn't impact the code correctness.